src/scripting/mapgen_lua_kernel.cpp

```diff
@@ -249,6 +249,18 @@ mapgen_lua_kernel::mapgen_lua_kernel(const config* vars)
 	lua_pop(L, 1);
 	assert(lua_gettop(L) == 0);
 
+	// Create the paths module
+	cmd_log_ << "Adding paths module...\n";
+	static luaL_Reg const path_callbacks[] {
+		{ "find_path",                 &intf_find_path                          },
```

Comment:
Although they have the same name, `game_lua_kernel::intf_find_path` and `mapgen_lua_kernel::intf_find_path` expect different argument lists. This seems a good time to give the mapgen one a different name.

Reply:
I thought one of those argument lists was sort of a subset of the other…? I'll take a closer look though.

Reply:
The difference pretty much amounts to an extra parameter in `mapgen_lua_kernel::intf_find_path` as well as it not supporting the options table (you must directly pass a function).

I think the two APIs could just be unified - add the extra options in `mapgen_lua_kernal` to `game_lua_kernel` and at least some of the extra options in `game_lua_kernel` to `mapgen_lua_kernel`. That may be out of scope for this PR, but I think I'll at least make it support an options table.